server/src/main/java/org/opensearch/search/profile/query/InternalProfileCollectorManager.java

```diff
@@ -82,7 +131,26 @@ public String getName() {
 
     @Override
     public CollectorResult getCollectorTree() {
-        return InternalProfileCollector.doGetCollectorTree(this);
+        return doGetCollectorManagerTree(this);
+    }
+
+    static CollectorResult doGetCollectorManagerTree(InternalProfileComponent collector) {
+        List<CollectorResult> childResults = new ArrayList<>(collector.children().size());
+        for (InternalProfileComponent child : collector.children()) {
+            CollectorResult result = doGetCollectorManagerTree(child);
```

Comment:
We can cast the children to be of type `InternalProfileCollectorManager` here as in `CollectorManager` tree thats what we will expect.

Reply:
Thanks, @sohami for the feedback. I have mad the necessary changes.